css-cascade-7/Overview.bs

```diff
@@ -0,0 +1,402 @@
+	</div>
+
+	<div class="example">
+		The following <a href="#conditional-import">conditional <css>@import</css> rule</a>
+		only loads the [=named stylesheet=] with a name of "foo" from "narrow.css" when the UA
+		<a href="https://www.w3.org/TR/css-conditional-3/#support-definition">supports</a> ''display: flex'',
+		and only applies the [=stylesheet=] on a <a href="https://www.w3.org/TR/CSS2/media.html#media-types">handheld</a> device
+		with a <a href="https://www.w3.org/TR/mediaqueries-4/#width">maximum viewport width</a> of 400px.
+
+		<pre>@import foo from url("narrow.css") supports(display: flex) handheld and (max-width: 400px);</pre>
+	</div>
+
+	<div class="example">
+		The following layer imports the [=named stylesheet=]s named "foo" and "bar" from "tabs.css" into
+		the ''framework.component'' layer, and an un-named layer, respectively:
+
+		<pre>
+		@import foo, bar from url("tabs.css") layer(framework.component);
+		@import foo, bar from url("override.css") layer;
+		</pre>
+	</div>
+
+	If a <url-string> is provided,
+	it must be interpreted as a <<url>> with the same value.
+
+	<div class="example">
+		The following lines are equivalent in meaning
+		and illustrate both ''@import'' syntaxes
+		(one with ''url()'' and one with a bare string):
+
+		<pre class='lang-css'>
+		@import "mystyle.css";
+		@import url("mystyle.css");
+		</pre>
+	</div>
+
+<!--
+ ██████   ██████   ██████   ███████  ██     ██
+██    ██ ██    ██ ██    ██ ██     ██ ███   ███
+██       ██       ██       ██     ██ ████ ████
+██        ██████   ██████  ██     ██ ██ ███ ██
+██             ██       ██ ██     ██ ██     ██
+██    ██ ██    ██ ██    ██ ██     ██ ██     ██
+ ██████   ██████   ██████   ███████  ██     ██
+-->
+
+<h2 id="sheet-apis">Sheet APIs</h2>
+
+<h3 id="the-csssheetblockrule-interface">
+The <code>CSSSheetBlockRule</code> interface</h3>
+
+	The {{CSSSheetBlockRule}} interface represents
+	the ''@sheet'' [[#at-ruledef-sheet|block rule]]:
+
+	<pre class='idl' export>
+	[Exposed=Window]
+	interface CSSSheetBlockRule : CSSGroupingRule {
+	  readonly attribute CSSOMString name;
+	};
+	</pre>
+
+	Its <dfn attribute for=CSSSheetBlockRule>name</dfn> attribute represents
+	the <<sheet-name>> declared by the at-rule itself.
+
+<h3 id="the-sheetnamelist-interface">The <code>SheetNameList</code> Interface</h3>
+
+An object that implements the {{SheetNameList}} interface 
+defines a list of stylesheet names.
+
+<pre class=idl>
+[Exposed=Window]
+interface SheetNameList {
+	readonly attribute unsigned long length;
+	getter CSSOMString? item(unsigned long index);
+};
+</pre>
+
+<h3 id="cssimportrule-extensions">
+Extensions to the <code>CSSImportRule</code> interface</h3>
+
+	The {{CSSImportRule}} interface is extended as follows:
+
+	<pre class='idl' export>
+	partial interface CSSImportRule {
+	  [SameObject] readonly attribute SheetNameList sheetNames;
+	  [SameObject] readonly attribute StyleSheetList namedStyleSheets;
```

Comment:
The value of `namedStyleSheets` should be manually available by iterating the existing `styleSheet` attribute and pulling out each `CSSSheetBlockRules` in order. Given that, is there a reason we should be exposing it as part of the standard? (Possibly an expected common use case?)

Reply:
You can manually access the individual rules that way, but not the actual `CSSStyleSheet` object.

The `CSSStyleSheet` has more options than the individual rules - https://drafts.csswg.org/cssom/#the-cssstylesheet-interface

This could potentially be plugged into `adoptedStyleSheets` in Shadow DOM as well - https://drafts.csswg.org/cssom/#dom-documentorshadowroot-adoptedstylesheets

(note that it currently says it must be a Constructable Stylesheet, but that was lifted recently and the specs/implementations haven't caught up - https://github.com/w3c/csswg-drafts/issues/10013#issuecomment-2165396092)

Reply:
Also, the `CSSImportRule` already has a single-value accessor for `styleSheet` - https://www.w3.org/TR/cssom-1/#cssimportrule

That won't work for named stylesheets, but separating them isn't too bad - the existing API will return the stylesheet containing the global, unnamed sheet and the new `namedStylesheets` list contains the `@sheet` nested sheets.